src/de/mod.rs

```diff
@@ -469,10 +469,10 @@ impl<'de, 'a> de::MapAccess<'de> for CommaSeparated<'a, 'de> {
         where K: DeserializeSeed<'de>
     {
         if self.has_element()? {
-            if self.terminator == b'}' {
-                seed.deserialize(&mut *self.de).map(Some)
+            if self.terminator == b')' {
```

Comment:
why is `MapAccess` separated by `)`?

Reply:
That's because structs are also considered maps; they share some common logic. That's just a check what it is we're deserializing in order to validate the correctness of the input.